packages/eslint-plugin-next/lib/rules/image-domain.js

```diff
@@ -0,0 +1,57 @@
+const { URL } = require('url')
+
+const { nextImage } = require('../utils/image.js')
+const NodeAttributes = require('../utils/nodeAttributes.js')
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Ensure domain or loader is configured for image optimization',
+      category: 'HTML',
+      recommended: true,
+    },
+  },
+  create: nextImage((context, node) => {
+    const [customNextConfigPath] = context.options
+
+    if (node.name.name !== 'Image') {
+      return
+    }
+
+    let attributes = new NodeAttributes(node)
+    let srcValue = attributes.value('src')
+
+    if (!srcValue.startsWith('http')) {
+      return
+    }
+
+    const imgHost = new URL(srcValue).hostname
+    const configFile =
+      customNextConfigPath || `${context.getCwd()}/next.config.js`
+
+    let domainConfigured = false
+    let loaderConfigured = false
+
+    try {
+      let config = require(configFile)
```

Comment:
This will break as it's not the only way to load next.config.js, you'll probably have to use the internal config loading mechanism for it.

Reply:
Removed the rule entirely for now. Using the same mechanism that `next` uses will make things a little tricker so I'll come back to this and submit as a separate PR :)